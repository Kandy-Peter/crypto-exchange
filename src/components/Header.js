import React from 'react';

const Header = () => {
  const padTo2Dgigits = (num) => num.toString().padStart(2, '0');
  const date = new Date();
  const FormatDate = () => [
    padTo2Dgigits(date.getDate()),
    padTo2Dgigits(date.getMonth() + 1),
    date.getFullYear(),
  ].join('/');

  return (
    <div className="header-container" data-testid="header-page">
      <div className="breaderHumb">
        Today&apos;s Crypto!
        <br />
        <span className="date"><FormatDate /></span>
      </div>
    </div>
  );
};
export default Header;
