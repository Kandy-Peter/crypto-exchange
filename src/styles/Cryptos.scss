@import './_colors';

.cryptos_row {
  .description {
    width: 100%;
    height: 60px;
    color: $gen_color;
    font-size: 16px;
    font-weight: 600;
    background-color: var(--bg-secondary);
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.5rem;
    box-shadow: 1px 2px 10px 5px #181c22;

    .search-box {
      background-color: #fefefe;
      height: 40px;
      display: flex;
      align-items: center;
      justify-content: center;
      cursor: pointer;
      padding: 12px;
      border-radius: 30px;
      box-shadow: 0 2px 10px #686868;

      input {
        width: 0;
        border: none;
        outline: none;
        transition: 0.5s;
        background-color: transparent;
      }

      &:hover input {
        width: 210px;
      }

      a #search-icon {
        display: flex;
        align-items: center;
        color: var(--background);
        font-size: 18px;
      }
    }
  }

  .all_cryptos {
    display: flex;
    flex-wrap: wrap;

    .container {
      color: var(--text-primary);
      width: 50%;
      font-size: 14px;
      text-align: center;
      padding: 0.4rem;

      &:nth-child(odd) {
        background-color: var(--background);
      }

      &:nth-child(even) {
        background-color: var(--card-color-2);
      }

      &:nth-child(4n-1) {
        background-color: var(--card-color-2);
      }

      &:nth-child(4n) {
        background-color: var(--background);
      }

      .card-row {
        width: 100%;
        display: flex;
        flex-direction: column;

        #arrow-icon {
          width: 100%;
          text-align: right;
          padding: 10px;
        }

        .img-clm {
          display: flex;
          justify-content: center;
          align-items: center;

          img {
            width: 80px;
            height: 80px;
            background-color: transparent;
          }
        }

        .crypto-card {
          margin-top: 0.6rem;

          .green_p {
            color: rgb(0, 129, 0);
          }

          .red_p {
            color: rgb(141, 0, 0);
            font-weight: 700;
          }
        }
      }
    }
  }
}
