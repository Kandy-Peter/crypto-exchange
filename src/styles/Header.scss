@import './_colors';

.header-container {
  color: var(--text-primary);
  width: 100%;

  .head {
    background: var(--background);
    height: 40px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 2rem;
    box-shadow: 1px 2px 10px 5px #181c22;

    h1,
    .icons {
      font-size: 22px;
    }

    .icons-menu {
      margin-left: 1.2rem;
    }
  }

  .breaderHumb {
    text-align: end;
    height: 200px;
    background: linear-gradient(270deg, var(--first-gradient) 49%, var(--second-gradient) 100%), url(../Images/background.jpg);
    background-size: cover;
    background-repeat: no-repeat;
    background-position: left;
    font-size: 22px;
    padding: 1.5rem;
    padding-top: 3rem;
    font-weight: 700;
    color: var(--text-primary);

    .date {
      font-size: 18px;
    }
  }
}
