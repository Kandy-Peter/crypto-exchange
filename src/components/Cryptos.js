import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaArrowCircleRight } from 'react-icons/fa';
import { RiSearchLine } from 'react-icons/ri';
import Header from './Header';
import Image from '../Images/Coins';

const Cryptos = () => {
  const cryptos = useSelector((state) => state.reducers.cryptos)
    .filter((coin) => coin.symbol.includes('USDT'));

  const [search, setSearch] = useState('');

  const matchValue = cryptos.filter((stock) => Object.keys(stock).some((key) => stock[key]
    .toString()
    .toLowerCase()
    .includes(search.toLocaleLowerCase())));

  const searchStock = (e) => {
    setSearch(e.target.value);
  };
  return (
    <div className="cryptos_row">
      <Header />
      <div className="description">
        Top 30:
        <div className="search-box">
          <input
            type="search"
            placeholder="Search Symbol..."
            onChange={searchStock}
            value={search}
          />
          <a href="#search">
            <RiSearchLine id="search-icon" />
          </a>
        </div>
      </div>
      <div className="all_cryptos">
        {matchValue.slice(0, 30).map((coin) => (
          <Link to={`coin/${coin.symbol}`} className="container" key={coin.symbol}>
            <div className="card-row">
              <span id="arrow-icon"><FaArrowCircleRight /></span>
              <span className="img-clm">
                <img src={Image(coin.symbol)} alt="crypto-img" />
              </span>
              <div className="crypto-card">
                <h4>{coin.symbol}</h4>
                <p className={coin.priceChangePercent > 0 ? 'green_p' : 'red_p'}>
                  {coin.priceChangePercent > 0 ? (`${coin.priceChangePercent}%`) : (`${coin.priceChangePercent}%`)}
                </p>
              </div>
            </div>
          </Link>
        ))}
      </div>
    </div>
  );
};

export default Cryptos;
