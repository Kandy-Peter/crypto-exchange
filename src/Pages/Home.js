import React from 'react';
import useLocalStorage from 'use-local-storage';
import { IoSettingsSharp } from 'react-icons/io5';
import { useSelector } from 'react-redux';
import { MdInvertColors } from 'react-icons/md';
import Cryptos from '../components/Cryptos';
import Loading from '../components/loader';

const Home = () => {
  const cryptos = useSelector((state) => state.reducers.cryptos);
  const defaultBlue = window.matchMedia('(prefers-color-scheme: blue)').matches;
  const [theme, setTheme] = useLocalStorage('theme', defaultBlue ? 'blue' : 'pink');

  const switchTheme = () => {
    const newTheme = theme === 'pink' ? 'blue' : 'pink';
    setTheme(newTheme);
  };

  return (
    <div className="header-container" data-theme={theme} data-testid="home-page">
      {!cryptos.length ? <Loading /> : (
        <>
          <div className="head">
            <h1>CoinTracker</h1>
            <div className="icons">
              <MdInvertColors className="icons-menu" onClick={switchTheme} />
              <IoSettingsSharp className="icons-menu" />
            </div>
          </div>
          <Cryptos />
        </>
      )}
    </div>
  );
};

export default Home;
