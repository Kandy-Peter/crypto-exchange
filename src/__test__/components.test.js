import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';

import Cryptos from '../components/Cryptos';
import Header from '../components/Header';
import Loading from '../components/loader';
import Home from '../Pages/Home';

afterEach(() => {
  cleanup();
});

window.matchMedia = window.matchMedia || function () {
  return {
    matches: false,
  };
};

const Mocked = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  </Provider>
);

describe('test the Home page', () => {
  test('check if the Home components it rendered correctly', () => {
    render(<Mocked />);
    const homePage = screen.getByTestId('home-page');
    expect(homePage).toBeTruthy();
  });

  test('test render snapshot for home page', () => {
    const tree = renderer.create(<Mocked />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test the Loading page', () => {
  it('test if the laoding is rendering correctly', () => {
    render(<Loading />);
    const loading = screen.getByTestId('loading-page');
    expect(loading).toBeTruthy();
  });

  test('test loading snapshot for loading page', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Loading />
        </BrowserRouter>
      </Provider>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('check if the Header is rendering correclty', () => {
  render(<Header />);
  const header = screen.getByTestId('header-page');
  expect(header).toBeTruthy();
});

it('Cryptos components it render correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Cryptos />
      </BrowserRouter>
    </Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
