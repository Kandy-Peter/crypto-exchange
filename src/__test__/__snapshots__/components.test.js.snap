// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cryptos components it render correctly 1`] = `
<div
  className="cryptos_row"
>
  <div
    className="loader"
    data-testid="loading-page"
  >
    <span
      className="line1"
    />
    <span
      className="line2"
    />
    <span
      className="line3"
    />
    <span
      className="line4"
    />
    <span
      className="line5"
    />
    <span
      className="line6"
    />
    <span
      className="line7"
    />
  </div>
</div>
`;

exports[`test the Home page test render snapshot for home page 1`] = `
<div
  className="header-container"
  data-testid="home-page"
  data-theme="pink"
>
  <div
    className="head"
  >
    <h1>
      CoinTracker
    </h1>
    <div
      className="icons"
    >
      <svg
        className="icons-menu"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M12 4.81V19c-3.31 0-6-2.63-6-5.87 0-1.56.62-3.03 1.75-4.14L12 4.81M6.35 7.56C4.9 8.99 4 10.96 4 13.13 4 17.48 7.58 21 12 21s8-3.52 8-7.87c0-2.17-.9-4.14-2.35-5.57L12 2 6.35 7.56z"
        />
      </svg>
      <svg
        className="icons-menu"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 176a80 80 0 1080 80 80.24 80.24 0 00-80-80zm172.72 80a165.53 165.53 0 01-1.64 22.34l48.69 38.12a11.59 11.59 0 012.63 14.78l-46.06 79.52a11.64 11.64 0 01-14.14 4.93l-57.25-23a176.56 176.56 0 01-38.82 22.67l-8.56 60.78a11.93 11.93 0 01-11.51 9.86h-92.12a12 12 0 01-11.51-9.53l-8.56-60.78A169.3 169.3 0 01151.05 393L93.8 416a11.64 11.64 0 01-14.14-4.92L33.6 331.57a11.59 11.59 0 012.63-14.78l48.69-38.12A174.58 174.58 0 0183.28 256a165.53 165.53 0 011.64-22.34l-48.69-38.12a11.59 11.59 0 01-2.63-14.78l46.06-79.52a11.64 11.64 0 0114.14-4.93l57.25 23a176.56 176.56 0 0138.82-22.67l8.56-60.78A11.93 11.93 0 01209.94 26h92.12a12 12 0 0111.51 9.53l8.56 60.78A169.3 169.3 0 01361 119l57.2-23a11.64 11.64 0 0114.14 4.92l46.06 79.52a11.59 11.59 0 01-2.63 14.78l-48.69 38.12a174.58 174.58 0 011.64 22.66z"
        />
      </svg>
    </div>
  </div>
  <div
    className="cryptos_row"
  >
    <div
      className="loader"
      data-testid="loading-page"
    >
      <span
        className="line1"
      />
      <span
        className="line2"
      />
      <span
        className="line3"
      />
      <span
        className="line4"
      />
      <span
        className="line5"
      />
      <span
        className="line6"
      />
      <span
        className="line7"
      />
    </div>
  </div>
</div>
`;

exports[`test the Loading page test loading snapshot for loading page 1`] = `
<div
  className="loader"
  data-testid="loading-page"
>
  <span
    className="line1"
  />
  <span
    className="line2"
  />
  <span
    className="line3"
  />
  <span
    className="line4"
  />
  <span
    className="line5"
  />
  <span
    className="line6"
  />
  <span
    className="line7"
  />
</div>
`;
